Release/src/http/client/http_client_winhttp.cpp

```diff
@@ -1053,6 +1069,112 @@ class winhttp_client : public _http_client_communicator
                     p_request_context->report_error(errorCode, build_error_msg(error_result));
                     break;
                 }
+			case WINHTTP_CALLBACK_STATUS_SENDING_REQUEST:
```

Comment:
This events comes twice, according to the documentation (https://msdn.microsoft.com/en-us/library/windows/desktop/aa384115%28v=vs.85%29.aspx): "Activates upon beginning and completing the sending of a request header with WinHttpSendRequest." 

This means we are doing pinning check twice for any HTTP requests - i'll see if there is any way to differentiate between the two occurences of the same callback.

Reply:
No, it's just once. The document you mentioned is for WINHTTP_CALLBACK_STATUS_SEND_REQUEST, which is defined as a combination of two events:
```c++
#define WINHTTP_CALLBACK_FLAG_SEND_REQUEST              (WINHTTP_CALLBACK_STATUS_SENDING_REQUEST | WINHTTP_CALLBACK_STATUS_REQUEST_SENT)
```